app/controllers/saved_searches_controller.rb

```diff
@@ -0,0 +1,39 @@
+# frozen_string_literal: true
+
+class SavedSearches::SavedSearchesController < ApplicationController
+  requires_plugin 'discourse-saved-searches'
+
+  def update
+    user = fetch_user_from_params
+    guardian.ensure_can_edit!(user)
```

Comment:
Should we also check whether the user being edited is allowed to have saved searches? Something like:

```
Guardian.new(user).ensure_can_use_saved_searches!
```